src/org/ods/OdsPipeline.groovy

```diff
@@ -75,10 +83,14 @@ class OdsPipeline implements Serializable {
 
             stages(context)
           }
-          script.currentBuild.result = 'SUCCESS'
+          if (context.displayNameUpdateEnabled) {
+            script.currentBuild.result = 'SUCCESS'
+          }
           setBitbucketBuildStatus('SUCCESSFUL')
         } catch (err) {
-          script.currentBuild.result = 'FAILURE'
+          if (context.displayNameUpdateEnabled) {
+            script.currentBuild.result = 'FAILURE'
+          }
```

Comment:
Seeing this I would extract the setting of the build result to a method that does the check.

However, why do we NOT want to set the result? I don't get that ...

Reply:
Extraxted. Should be handled by enclosing pipeline.